pkg/asset/machines/machineconfig/hyperthreading.go

```diff
@@ -0,0 +1,40 @@
+package machineconfig
+
+import (
+	"fmt"
+
+	igntypes "github.com/coreos/ignition/config/v2_2/types"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset/ignition"
+)
+
+// ForHyperthreadingDisabled creates the MachineConfig to enable hyperthreading.
+// RHCOS ships with pivot.service that uses the `/etc/pivot/kernel-args` to override the kernel arguments for hosts.
+func ForHyperthreadingDisabled(role string) *mcfgv1.MachineConfig {
+	return &mcfgv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "machineconfiguration.openshift.io/v1",
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("99-%s-hyperthreading", role),
+			Labels: map[string]string{
+				"machineconfiguration.openshift.io/role": role,
+			},
+		},
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: igntypes.Config{
+				Ignition: igntypes.Ignition{
+					Version: igntypes.MaxVersion.String(),
+				},
+				Storage: igntypes.Storage{
+					Files: []igntypes.File{
+						ignition.FileFromString("/etc/pivot/kernel-args", "root", 0600, "ADD nosmt"),
```

Comment:
:+1:

Reply:
How are these assembled if other folks add MachineConfigs with their own kernel options?  Do we need to set `Append` on this?

Reply:
`/etc/pivot/kernel-args` is a RHCOS impl and it only accepts the `ADD nosmt` so no append.

Reply:
> `/etc/pivot/kernel-args` is a RHCOS impl and it only accepts the `ADD nosmt` so no append.

Oh, so there's no ability for `ADD some-other-flag` at the moment?

Reply:
If someone attempts to `ADD` or `DELETE` any other argument it will skip it and log. Only items in the whitelist inside `pivot` are allowed.

Reply:
https://github.com/openshift/pivot/blob/master/cmd/root.go#L43-L45